utils/common/src/main/kotlin/Utils.kt

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.utils.common
+
+import java.io.File
+
+/**
+ * Return recursively all ancestor directories of the given absolute [file], ordered along the path from
+ * the parent of [file] to the root.
+ */
+fun getAllAncestorDirectories(file: String): List<String> {
+    val result = mutableListOf<String>()
+
+    var ancestorDir = File(file).parentFile
+    while (ancestorDir != null) {
+        result += ancestorDir.invariantSeparatorsPath
+        ancestorDir = ancestorDir.parentFile
+    }
+
+    return result
+}
+
+/**
+ * Return the longest parent directory that is common to all [files], or null if they have no directory in common.
+ */
+fun getCommonFileParent(files: Collection<File>): File? =
+    files.map {
+        it.normalize().absolutePath
+    }.reduceOrNull { prefix, path ->
+        prefix.commonPrefixWith(path)
+    }?.let {
+        val commonPrefix = File(it)
+        if (commonPrefix.isDirectory) commonPrefix else commonPrefix.parentFile
+    }
+
+/**
+ * Return the full path to the given executable file if it is in the system's PATH environment, or null otherwise.
+ */
+fun getPathFromEnvironment(executable: String): File? {
```

Comment:
Did we want to move this (and `resolveWindowsExecutable`) to inside `Os`?

Reply:
Yes, I wanted to do this in the next PR because there were some open questions which I wanted to discuss with you first.